playbooks/roles/nghttpx/tasks/install.yml

```diff
@@ -0,0 +1,28 @@
+---
+- name: Install the nghttpx dependencies
+  apt:
+    name: "{{ item }}"
+  with_items: "{{ nghttpx_dependencies }}"
+
+- name: Download the nghttpx source code
+  get_url:
+    url: "{{ nghttpx_download_url }}"
+    dest: "{{ nghttpx_source_location }}"
+    checksum: "{{ nghttpx_source_checksum }}"
+    mode: 0644
+
+- name: Extract the nghttpx source code
+  unarchive:
+    remote_src: yes
+    src: "{{ nghttpx_source_location }}"
+    dest: /usr/local/src
+    owner: root
+    group: root
+    creates: "{{ nghttpx_compilation_directory }}/README"
+
+- name: Compile and install nghttpx
+  shell: ./configure && make -j {{ ansible_processor_cores }} && make install && install -Dm755 contrib/nghttpx.service /lib/systemd/system/nghttpx.service
```

Comment:
Yes, much better to install from package where possible because that way the unattended upgrades configuration ensures they get security updates automatically. With source compiled software if you install Streisand today, you will be on that version forever. Nothing updates the source built components down the road for new patches/fixes.